File: db4o.net/Db4objects.Db4o/Db4objects.Db4o/Internal/Query/NQOptimizationInfo.cs
/* Copyright (C) 2004 - 2011  Versant Inc.  http://www.db4o.com */

namespace Db4objects.Db4o.Internal.Query
{
	public class NQOptimizationInfo
	{
		private Db4objects.Db4o.Query.Predicate _predicate;

		private string _message;

		private object _optimized;

		public NQOptimizationInfo(Db4objects.Db4o.Query.Predicate predicate, string message
			, object optimized)
		{
			this._predicate = predicate;
			this._message = message;
			this._optimized = optimized;
		}

		public virtual string Message()
		{
			return _message;
		}

		public virtual object Optimized()
		{
			return _optimized;
		}

		public virtual Db4objects.Db4o.Query.Predicate Predicate()
		{
			return _predicate;
		}

		public override string ToString()
		{
			return Message() + "/" + Optimized();
		}
	}
}

File: db4oj/core/src/com/db4o/internal/query/NQOptimizationInfo.java
/* Copyright (C) 2004 - 2006  Versant Inc.  http://www.db4o.com */

package com.db4o.internal.query;

import com.db4o.query.*;

public class NQOptimizationInfo {
	private Predicate _predicate;
	private String _message;
	private Object _optimized;

	public NQOptimizationInfo(Predicate predicate, String message, Object optimized) {
		this._predicate = predicate;
		this._message = message;
		this._optimized = optimized;
	}

	public String message() {
		return _message;
	}

	public Object optimized() {
		return _optimized;
	}

	public Predicate predicate() {
		return _predicate;
	}

	public String toString() {
		return message()+"/"+optimized();
	}
}
